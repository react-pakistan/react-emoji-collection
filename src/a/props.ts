import { theme } from '@taimoormk/util-functions';
import { text } from '@storybook/addon-knobs';
import { IAProps } from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const aProps = () : IAProps => ({
  color: text('color', theme.colors.black),
  desktop: text('desktop', desktop),
  download: text('download', ''),
  href: text('href', 'https://google.com'),
  hreflang: text('hreflang', ''),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  media: text('media', ''),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  referrerpolicy: text('referrerpolicy', ''),
  rel: text('rel', ''),
  tablet: text('tablet', tablet),
  target: text('target', '_blank'),
  type: text('type', ''),
});
