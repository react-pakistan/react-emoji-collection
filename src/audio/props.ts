import { boolean, text } from '@storybook/addon-knobs';
import { theme } from '@taimoormk/util-functions';
import { IAudioProps } from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const audioProps = () : IAudioProps => ({
  autoplay: boolean('autoplay', true),
  controlPanelColor: text('controlPanelColor', theme.colors.lightGrey),
  controls: boolean('controls', true),
  currentTimeColor: text('currentTimeColor', theme.colors.paleGreen),
  desktop: text('desktop', desktop),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  loop: boolean('loop', false),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  muted: boolean('muted', false),
  preload: text('preload', 'auto'),
  remainingTimeColor: text('remainingTimeColor', theme.colors.cherryRed),
  src: text('src', 'https://notificationsounds.com/sound-effects/insight-578/download/mp3'),
  tablet: text('tablet', tablet),
});
