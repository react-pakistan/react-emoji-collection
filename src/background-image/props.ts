import { text } from '@storybook/addon-knobs';
import { IBackgroundImageProps } from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const backgroundImageProps = () : IBackgroundImageProps => ({
  desktop: text('desktop', desktop),
  imageBackgroundColor: text('imageBackgroundColor', '#6C6C6C'),
  imageBackgroundHeight: text('imageBackgroundHeight', '400px'),
  imageBackgroundMargin: text('imageBackgroundMargin', '0'),
  imageBackgroundPadding: text('imageBackgroundPadding', '0'),
  imageBackgroundPosition: text('imageBackgroundPosition', 'center center'),
  imageBackgroundRepeat: text('imageBackgroundRepeat', 'no-repeat'),
  imageBackgroundSize: text('imageBackgroundSize', 'cover'),
  imageBackgroundWidth: text('imageBackgroundWidth', '600px'),
  imageBorderRadius: text('imageBorderRadius', '0'),
  imageBackgroundUrl: text(
    'imageBackgroundUrl',
    'https://cdn-images-1.medium.com/max/1200/1*y6C4nSvy2Woe0m7bWEn4BA.png',
  ),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  tablet: text('tablet', tablet),
});
