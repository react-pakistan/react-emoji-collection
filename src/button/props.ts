import {
  boolean,
  select,
  text,
} from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { theme } from '@taimoormk/util-functions';
import {
  BUTTON_FORM_ENCTYPE,
  BUTTON_FORM_METHODS,
  BUTTON_FORM_TARGET,
  BUTTON_TYPE,
  IButtonAttrsProps,
  IButtonProps,
} from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const buttonProps = () : IButtonAttrsProps & IButtonProps => ({
  autoFocus: boolean('autoFocus', true),
  border: text('border', `1px solid ${theme.colors.darkGrey}`),
  borderRadius: text('borderRadius', '0.5em'),
  backgroundColor: text('backgroundColor', theme.colors.white),
  desktop: text('desktop', desktop),
  disabled: boolean('disabled', false),
  form: text('form', ''),
  formaction: text('formaction', ''),
  formenctype: select('formenctype', {
    application: BUTTON_FORM_ENCTYPE.APPLICATION,
    multipart: BUTTON_FORM_ENCTYPE.MULTIPART,
    text: BUTTON_FORM_ENCTYPE.TEXT,
  }, BUTTON_FORM_ENCTYPE.TEXT),
  formmethod: select('formmethod', {
    get: BUTTON_FORM_METHODS.GET,
    post: BUTTON_FORM_METHODS.POST,
  }, BUTTON_FORM_METHODS.GET),
  formnovalidate: boolean('formnovalidate', false),
  formtarget: select('formtarget', {
    blank: BUTTON_FORM_TARGET.BLANK,
    parent: BUTTON_FORM_TARGET.PARENT,
    self: BUTTON_FORM_TARGET.SELF,
    top: BUTTON_FORM_TARGET.TOP,
  }, BUTTON_FORM_TARGET.SELF),
  height: text('height', 'auto'),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  name: text('name', ''),
  onClick: action('button clicked'),
  padding: text('padding', '1em'),
  tablet: text('tablet', tablet),
  type: select('type', {
    button: BUTTON_TYPE.BUTTON,
    reset: BUTTON_TYPE.RESET,
    submit: BUTTON_TYPE.SUBMIT,
  }, BUTTON_TYPE.BUTTON),
  value: text('value', ''),
  width: text('width', 'auto'),
});
