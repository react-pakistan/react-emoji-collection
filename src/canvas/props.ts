import { text } from '@storybook/addon-knobs';
import {
  ICanvasAttrsProps,
  ICanvasProps,
} from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const canvasProps = () : ICanvasAttrsProps & ICanvasProps => ({
  border: text('border', '1px solid #000000'),
  desktop: text('desktop', desktop),
  height: text('height', '400px'),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  id: text('id', 'canvasId'),
  tablet: text('tablet', tablet),
  width: text('width', '600px'),
});
