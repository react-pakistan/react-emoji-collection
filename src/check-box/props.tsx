import { boolean, text } from '@storybook/addon-knobs';
import { IconCheck } from '@taimoormk/react-icon-collection/icons/misc-grey';
import { theme } from '@taimoormk/util-functions';
import React from 'react';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const checkBoxProps = () : any => ({
  check: <IconCheck fill={theme.colors.white} fontSize='20px' />,
  checkedColor: text('checkedColor', theme.colors.skyBlue),
  desktop: text('desktop', desktop),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  tablet: text('tablet', tablet),
  unCheckedColor: text('unCheckedColor', theme.colors.lighterBlack),
});

export const checkBoxWithLabelProps = () : any => ({
  check: <IconCheck fill={theme.colors.white} fontSize='20px' />,
  checkedColor: text('checkedColor', theme.colors.skyBlue),
  desktop: text('desktop', desktop),
  hasError: boolean('hasError', false),
  helperText: text('helperText', 'There is an error'),
  helperTextColor: text('helperTextColor', theme.colors.cherryRed),
  label: text('label', 'Pick one'),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  tablet: text('tablet', tablet),
  unCheckedColor: text('unCheckedColor', theme.colors.lighterBlack),
});
