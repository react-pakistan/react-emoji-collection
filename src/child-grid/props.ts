import { number, text } from '@storybook/addon-knobs';
import { IChildGridProps } from '..';

export const mobileS = '';

export const mobileM = '';

export const mobileL = '';

export const tablet = '';

export const laptop = '';

export const laptopL = '';

export const desktop = '';

export const childGridProps = () : IChildGridProps => ({
  alignSelf: text('alignSelf', 'center'),
  desktop: text('desktop', desktop),
  gridColumnEnd: number('gridColumnEnd', 1),
  gridColumnStart: number('gridColumnStart', 1),
  gridRowEnd: number('gridRowEnd', 1),
  gridRowStart: number('gridRowStart', 1),
  justifySelf: text('justifySelf', 'center'),
  laptop: text('laptop', laptop),
  laptopL: text('laptopL', laptopL),
  mobileL: text('mobileL', mobileL),
  mobileM: text('mobileM', mobileM),
  mobileS: text('mobileS', mobileS),
  tablet: text('tablet', tablet),
});
