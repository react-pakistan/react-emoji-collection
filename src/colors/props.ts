import { theme } from '@taimoormk/util-functions';
import { object } from '@storybook/addon-knobs';
import { IColorsProps } from '..';

export const colorsProps = () : IColorsProps => ({
  colors: object('colors', theme.colors),
});
