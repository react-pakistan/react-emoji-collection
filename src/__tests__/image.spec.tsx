import React from 'react';
import { mount } from 'enzyme';
import { StyledStory } from '../styled-app';
import { Image } from '../image';
import { imageProps } from '../image/props';

describe('Testing Image component', () => {
  it('matches Image snapshot', () => {
    const wrapper = mount(
      <StyledStory>
        <Image {...imageProps()} />
      </StyledStory>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
