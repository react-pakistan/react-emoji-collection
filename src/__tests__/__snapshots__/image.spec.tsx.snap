// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Image component matches Image snapshot 1`] = `ReactWrapper {}`;
